pkgs/tools/misc/scriptcommunicator/default.nix

```diff
@@ -0,0 +1,93 @@
+{ stdenv
+, fetchFromGitHub, qmake, pkgconfig
+, libpng, zlib, unzip
+, qt5, qttools, qtserialport, qtscript, qtmultimedia
+, makeWrapper, makeDesktopItem
+}:
+
+let
+  pname = "scriptcommunicator";
+  version = "05.10";
+
+  desktopItem = makeDesktopItem {
+    name = "scriptcommunicator";
+    exec = "${pname}";
+    icon = "$out/share/scriptcommunicator/images/main_32x32.png";
+    comment = "Scriptable data terminal"; 
+    desktopName = "scriptcommunicator";
+    genericName = "scriptcommunicator";
+    categories = "Network;";
+  };
+
+in stdenv.mkDerivation rec {
+
+  inherit pname version;
+
+  nativeBuildInputs = [ unzip qmake makeWrapper ];
+  buildInputs = [  qtserialport qtscript qtmultimedia ];
+
+  # we want QtDesigner available to create UI
+  propagatedBuildInputs = [ qttools ]; 
+
+  src = fetchFromGitHub {
+    owner = "szieke";
+    repo = "ScriptCommunicator_serial-terminal";
+    rev = "Release_05_10";
+    sha256 = "0v6lh5di5pxkqklp3iyvdqmr7w66c7xdd1r8lkk43sh7w7jnbvc9";
+  };
+
+  sourceRoot = "source/ScriptCommunicator";
+
+  buildPhase = ''
+    cd DeleteFolder/DeleteFolder
```

Comment:
Do you have to `cd` into the directories for `qmake` to work?
If you have to, I think you could use `qmake -r DeleteFolder.pro ScriptEditor.pro ScriptCommunicator.pro`

Reply:
No way, i don't success to do that.

Reply:
`pushd` and `popd` might be a bit nicer to use here

Reply:
@bignaux ping

Reply:
i don't have time to make a perfect recipe for this bugging tool. I improve derivation on new release, time to time. Better is the enemy of good.

Reply:
@bignaux no worries. Just trying to be helpful and promote consistent coding style within nixpkgs.